test/v2/controllers/requestParsers/validators/CreateForeignPropertyPeriodSummaryValidatorSpec.scala

```diff
@@ -460,6 +460,64 @@ class CreateForeignPropertyPeriodSummaryValidatorSpec extends UnitSpec with Mock
             )
           )) shouldBe List(FromDateFormatError)
       }
+      "the fromDate is out of range" in {
+        val validator = setUpValidator()
+        validator.validate(
+          CreateForeignPropertyPeriodSummaryRawData(
+            validNino,
+            validBusinessId,
+            validTaxYear,
+            Json.parse(
+              """
+                |{
+                |   "fromDate":"1782-01-01",
+                |   "toDate":"2021-03-29",
+                |   "foreignFhlEea":{
+                |      "income":{
+                |         "rentAmount":381.21
+                |      },
+                |      "expenses":{
+                |         "premisesRunningCosts":993.31,
+                |         "repairsAndMaintenance":8842.23,
+                |         "financialCosts":994,
+                |         "professionalFees":992.12,
+                |         "costOfServices":4620.23,
+                |         "travelCosts":774,
+                |         "other":984.41
+                |      }
+                |   },
+                |   "foreignNonFhlProperty":[
+                |      {
+                |         "countryCode":"AFG",
+                |         "income":{
+                |            "rentIncome":{
+                |               "rentAmount":4882.23
+                |            },
+                |            "foreignTaxCreditRelief":true,
+                |            "premiumsOfLeaseGrant":884.72,
+                |            "otherPropertyIncome":7713.09,
+                |            "foreignTaxPaidOrDeducted":884.12,
+                |            "specialWithholdingTaxOrUkTaxPaid":847.72
+                |         },
+                |         "expenses":{
+                |            "premisesRunningCosts":129.35,
+                |            "repairsAndMaintenance":7490.32,
+                |            "financialCosts":5000.99,
+                |            "professionalFees":847.90,
+                |            "travelCosts":69.20,
+                |            "costOfServices":478.23,
+                |            "residentialFinancialCost":879.28,
+                |            "broughtFwdResidentialFinancialCost":846.13,
+                |            "other":138.92
+                |         }
+                |      }
+                |   ]
+                |}
+                |""".stripMargin
+            )
+          )) shouldBe List(FromDateOutOfRangeError)
```

Comment:
Just a stylistic/convention thing, but could the result be assigned to a "val result"?  Then the final line is "result shouldBe ..."

Reply:
@molafisoye I noticed the json used for test is used in multiple places, and only thing that is changing is mainly date or  some other attribute. Can we not extract common json and just replace the attribute we need to modify. I know this is not related to this test, but just by looking at Json it doesn't clearly tell me what all attributes are changing.